aten/src/ATen/native/ConvUtils.h

```diff
@@ -106,4 +107,31 @@ static inline bool cudnn_conv_use_channels_last(const at::Tensor& input, const a
   return can_use_cudnn_channels_last_2d || can_use_cudnn_channels_last_3d;
 }
 
+static inline bool miopen_conv_use_channels_last(const at::Tensor& input, const at::Tensor& weight) {
+
+  static c10::optional<bool> PYTORCH_MIOPEN_SUGGEST_NHWC = c10::utils::check_env("PYTORCH_MIOPEN_SUGGEST_NHWC");
+
+  // disable NHWC for float64 input.
+  if (!at::detail::getCUDAHooks().compiledWithMIOpen() ||
+      input.scalar_type() == at::kDouble ||
+      weight.scalar_type() == at::kDouble) {
+    return false;
+  }
+
+  auto input_memory_format = input.suggest_memory_format();
+  auto weight_memory_format = weight.suggest_memory_format();
+
+  bool can_use_miopen_channels_last_2d = false;
+#if defined(USE_ROCM) && (ROCM_VERSION >= 40300)
+  can_use_miopen_channels_last_2d = PYTORCH_MIOPEN_SUGGEST_NHWC &&  *PYTORCH_MIOPEN_SUGGEST_NHWC && (
```

Comment:
What happens when the optional<bool> is a nullopt because the user set the env var to something other than 0 or 1?  Do we care about that use case and failing gracefully?

Aside: The c10::utils::check_env appears to have been added recently (#59052), but is currently not used anywhere else.  ATen seems to use getenv directly.  Does upstream have a preference how to parse env vars?

Reply:
You can see that I am testing both `PYTORCH_MIOPEN_SUGGEST_NHWC` and `*PYTORCH_MIOPEN_SUGGEST_NHWC`. The first one is the nullopt check. The second is the true/false check. About handling other arguments like 'True/False', thats upto check_env. 

Yes, c10::utils::check_env is new, and not used. I figured I will use it since the API is supported, and it might be a direction where the PyTorch devs want to go.

Reply:
sorry, I missed the nullopt check.  looking forward to hearing from upstream reviewer how they'd like to handle env vars.

Reply:
Can you explain why this feature has to be gated by an environment variable. For alternative layout support, ordinarily you would simply just do the correct layout algorithm depending on what the layout of the weights are. To compare, cuDNN supports alternative layouts without needing an envvar to modulate.

Reply:
The reason it has to be gated is because MIOpen does not officially support NHWC yet, although convolution support has been added to tip-of-tree of MIOpen. The plan is to remove the environment variable check to a ROCm version check once support is officially added. 
( Also, the MIOpen teams do need this support in PyTorch during this time, since we are testing performance of the newly added NHWC code on application loads. )

Reply:
OK, but it's pretty difficult to "accidentally" end up running NHWC when you didn't intend to (the weights have to be NHWC). Wouldn't you just rather error in that case for now?

Reply:
In any case, the plan on record was not clear from the PR description nor the code